src/middlewared/middlewared/utils/interface.py

```diff
@@ -0,0 +1,40 @@
+import contextlib
+import os
+import time
+
+
+IFACE_LINK_STATE_MAX_WAIT: int = 60
+
+
+def get_default_interface() -> str | None:
+    data = []
+    with contextlib.suppress(FileNotFoundError):
+        with open('/proc/net/route', 'r') as f:
+            data = [line.split() for line in f.readlines()]
```

Comment:
This is a bit careless and looking at route.h in linux sources, we can determine the default gateway a bit more "properly".
```
RT_UP = 0x0001
RT_GATEWAY = 0x0002
with open('/proc/net/route') as f:
    for lineno, line in filter(lambda x: x[0] > 1, f):
        if (columns := line.strip().split()) and columns[3] == RTF_UP | RTF_GATEWAY:
                return columns[0].strip()